### PR TITLE
For loops in Scala

### PR DESCRIPTION
**Pull request by cankill**
Pull request for task: http://jira.baeldung.com/browse/SCALA-16

### CODE REVIEW

**Comment by cankill** [comment]
@jzheaux, please check again, I fixed the names.

File: core-scala/src/test/scala/com/baeldung/scala/forcomprehension/ForLoopTest.scala [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  override def sideEffectFunction (arg: Any): Unit = {
+    sideEffectResult = sideEffectResult :+ arg
+  }
+
+  def clean(): Unit = {
+    sideEffectResult = Nil;
+  }
+}
+
+class ForLoopTest {
```
Comment (KevinGilmore) [inline core-scala/src/test/scala/com/baeldung/scala/forcomprehension/ForLoopTest.scala]: It seems that the *UnitTest naming convention applies for the Scala repo as well -- all the current test class files are named that way.

**Comment by cankill** [comment]
@KevinGilmore, please check, is it ok now?

**Comment by cankill** [comment]
@jzheaux, please check.

**Pull request merged**